java/core/src/test/java/com/lancedb/lance/VectorSearchTest.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lancedb.lance;
+
+import org.apache.arrow.dataset.scanner.Scanner;
+import org.apache.arrow.vector.Float4Vector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import com.lancedb.lance.ipc.Query;
+import com.lancedb.lance.ipc.ScanOptions;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+// Creates a dataset with 5 batches where each batch has 80 rows
+//
+// The dataset has the following columns:
+//
+//  i   - i32      : [0, 1, ..., 399]
+//  s   - &str     : ["s-0", "s-1", ..., "s-399"]
+//  vec - [f32; 32]: [[0, 1, ... 31], [32, ..., 63], ... [..., (80 * 5 * 32) - 1]]
+//
+// An IVF-PQ index with 2 partitions is trained on this data
+public class VectorSearchTest {
+  @TempDir
+  Path tempDir;
+
+  @Test
+  void test_create_index() throws Exception {
```

Comment:
@walterddr This is the search unit test and this file contains the search-related unit tests.

Lance supported creating different indexes. Those indexes help improve the performance, but does not affect user behavior.
Two ways to identity the index is created:
1. list indexes to get index information
2. validate the projection plan gets changed to involve using different indexes
https://github.com/lancedb/lance/blob/ef0953df8220a36b6433e47bea6112ce4fc7f3c8/rust/lance/src/dataset/scanner.rs#L4058

Reply:
i was wondering if we have tests similar to https://lancedb.github.io/lancedb/search/#__tabbed_3_1